tests/fixtures/testModel.ts

```diff
@@ -136,6 +138,24 @@ export interface TestModel extends Model {
         [name: string]: TypeAliasModel1;
       };
     };
+    /** @deprecated */
+    deprecatedSubProperty?: number;
+  };
+
+  /** not deprecated */
+  notDeprecatedProperty?: number;
+  /** type is deprecated, but this property is not */
```

Comment:
I'm wondering if i should rethink this, as I'm right in the middle of adding better `deprecated` support to openapi-generator and this seems at odds with that... In OAS terms, this becomes a `ref`, and if that corresponding schema happens to be deprecated, then this property should be considered deprecated, as everything from the ref comes in.

By the same token I need a test explicitly around deprecating a type alias (like line 141, but not a primitive); the ref overrides everything including an explicit `deprecated`. Might need to give that some thought (perhaps emit a warning?)

Reply:
Here's what I've previously done, I think it was w/ regards to `descriptions`s along $refs. We still add it to the spec, knowing whoever parses the spec is bound by the spec to treat this as: `Any members [other than "$ref] SHALL be ignored.`. Warning about that seems reasonable, but we don't currently do it.

Seems like was a reasonable choice given that OpenAPI 3.1 allows for this explicitly: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#referenceObject